api/controllers/SearchController.js

```diff
@@ -1,11 +1,21 @@
 var findCommunityIds = Promise.method(req => {
   if (req.param('communityId')) {
     return [req.param('communityId')]
+  } else if (req.param('type') === 'communities' && req.param('moderated') && Admin.isSignedIn(req)) {
+    return Community.fetchAll()
+    .then(cs => Promise.map(cs.models, c => c.id))
   } else {
     return Promise.join(
       Network.activeCommunityIds(req.session.userId),
       Membership.activeCommunityIds(req.session.userId)
-    ).then(ids => _(ids).flatten().uniq().value())
+    ).then(dupIds => {
+      var ids = _(dupIds).flatten().uniq().value()
+      if (req.param('moderated')) {
+        return Promise.filter(ids, id => Membership.hasModeratorRole(req.session.userId, id))
```

Comment:
that's never going to run for Ed though, because he'll be logged in as Admin, so it runs through the
`req.param('type')` === 'communities' && req.param('moderated') && Admin.isSignedIn(req)
branch

Still want me to write the new method?